TRACING.md

```diff
@@ -0,0 +1,312 @@
+   [osbuilder](https://github.com/kata-containers/osbuilder).
+
+   - The image **must** include:
+     - The `jaeger-client-socat-redirector.service` systemd service
+       (see the [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock) section).
+     - The distribution-specific package including the `socat(1)` utility
+       (normally the package is called `socat`).
+
+   - The image can include:
+     - The `kata-journald-host-redirect.service` service
+       (see the [Guest journal redirector](#guest-journal-redirector) section).
+     - The `redirect-agent-output-to-journal.conf` systemd snippet
+       (see the [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock) section).
+
+1. Install the custom image.
+
+1. Configure the runtime to:
+
+   - Use the custom image by modifying the `image=` option in the runtime's `configuration.toml` file.
+   - Set the correct kernel parameters for agent tracing. See:
+
+     - The [Tracing options](#tracing-options) section.
+
+     - If you also want full debug and guest journal redirection, see the
+       [Enable full agent debug](#enable-full-agent-debug) section.
+
+1. Install and run Jaeger on the host.
+
+   The simplest method to start using Jaeger is to follow the instructions to run the
+   ["all-in-one" Docker\* image](https://www.jaegertracing.io/docs/1.8/getting-started/#all-in-one).
+
+1. Start the VSOCK Jaeger redirector script on the host:
+
+   ```
+   $ sudo scripts/vsock-to-udp-server.sh
+   ```
+
+1. Start the journal redirector script on the host (OPTIONAL):
+
+   ```
+   $ sudo scripts/vsock-to-text-server.sh
+   ```
+
+1. Create a Kata Container.
+
+1. Stop the Kata Container.
+
+1. View the [Jaeger UI](https://www.jaegertracing.io/docs/getting-started).
+
+# Requirements
+
+## Host environment
+
+- Host kernel must support the VSOCK socket type:
+
+  kernel config option: `CONFIG_VHOST_VSOCK`.
+
+- VSOCK kernel module must be loaded:
+
+   ```
+   $ sudo modprobe vhost_vsock
+   ```
+
+- VSOCK support must be enabled in the `kata-runtime`'s `configuration.toml`
+  config file:
+
+  ```
+  $ sudo sed 's/#use_vsock =.*/use_vsock = true/g' /usr/share/defaults/kata-containers/configuration.toml
+  ```
+
+## Guest environment
+
+- Guest kernel must support the VSOCK socket type:
+
+  kernel config option: `CONFIG_VIRTIO_VSOCKETS`.
+
+# Limitations
+
+- The image configured in the `kata-runtime` `configuration.toml` file must
+  be specified using `image=` (`initrd=` is not supported).
+
+  This is due to having to run additional systemd services inside the VM
+  currently, to proxy the Jaeger trace flows out to the Jaeger collector running
+  on the host.
+
+  See the
+  [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock)
+  section for further details.
+
+- Tracing is only finalised when the workload has exited
+
+  Although trace *can* be inspected before the workload exits, it will be
```

Comment:
Suggested rewrite to stay active: Although trace *can* be inspected before the workload exits, it is incomplete.